### PR TITLE
Fixed #26107 -- Added option to int_list_validator to allow negatives

### PR DESCRIPTION
**Pull request by akki**
The new boolean allow_negative option will let the list contain negative
 integers.

### CODE REVIEW

File: django/core/validators.py [hunk 1]
```diff
@@ -283,8 +283,11 @@ def ip_address_validators(protocol, unpack_ipv4):
                          % (protocol, list(ip_address_validator_map)))
 
 
-def int_list_validator(sep=',', message=None, code='invalid'):
-    regexp = _lazy_re_compile('^\d+(?:%s\d+)*\Z' % re.escape(sep))
+def int_list_validator(sep=',', message=None, code='invalid', allow_negative=False):
+    if allow_negative is True:
+        regexp = _lazy_re_compile('^(-)?\d+(?:%s(-)?\d+)*\Z' % re.escape(sep))
```
Comment (timgraham) [inline django/core/validators.py]: I'd rather build regex string with interpolation so as not to repeat the other logic.

File: docs/ref/validators.txt [hunk 2]
```diff
@@ -222,12 +222,13 @@ to, or in lieu of custom ``field.clean()`` methods.
...
 ----------------------
 
-.. function:: int_list_validator(sep=',', message=None, code='invalid')
+.. function:: int_list_validator(sep=',', message=None, code='invalid', allow_negative=False)
 
     .. versionadded:: 1.9
 
     Returns a :class:`RegexValidator` instance that ensures a string
-    consists of integers separated by ``sep``.
+    consists of integers separated by ``sep``
```
Comment (timgraham) [inline docs/ref/validators.txt]: . It allows.... (wrap to previous line)
Also need a `.. versionchanged:: 1.10` annotation and a mention in the 1.10 release notes.

File: django/core/validators.py (hunk 1 above)
Comment (akki) [inline django/core/validators.py]: Do you mean something like this ?
```python
if allow_negative is True:
    regex_string = '^(-)?\d+(?:%s(-)?\d+)*\Z' % re.escape(sep)
elif allow_negative is True:
    regex_string = '^\d+(?:%s\d+)*\Z' % re.escape(sep)

regexp = _lazy_re_compile(regex_string)
```

Comment (timgraham) [inline django/core/validators.py]: more like: `'^%(negative)s\d+(?:%(sep)s%(negative)s?\d+)*\Z'  % ({'negative': '(-)?' if allow_negative else '', ...})`

File: docs/ref/validators.txt [hunk 3]
```diff
@@ -222,12 +222,15 @@ to, or in lieu of custom ``field.clean()`` methods.
...
+.. function:: int_list_validator(sep=',', message=None, code='invalid', allow_negative=False)
 
     .. versionadded:: 1.9
 
+    .. versionchanged:: 1.10
+
     Returns a :class:`RegexValidator` instance that ensures a string
-    consists of integers separated by ``sep``.
+    consists of integers separated by ``sep``. It allows negative integers when
+    ``allow_negative`` is ``True``.
```
Comment (akki) [inline docs/ref/validators.txt]: I am a bit confused whether I should write the new feature and the existing class description 1) separately under the versionadded and versionchanged block or
2) the way I have done it right now.

**Comment by akki** [comment]
I have made the suggested changes. Please have a look now.

File: docs/releases/1.10.txt [hunk 4]
```diff
@@ -351,6 +351,10 @@ Validators
   domain name labels to 63 characters and the total length of domain
   names to 253 characters per :rfc:`1034`.
 
+* :class:`~django.core.validators.int_list_validator` now accepts an optional
```
Comment (Brobin) [inline docs/releases/1.10.txt]: I think `:class:` should be `:func:`.

File: docs/ref/validators.txt (hunk 3 above)
Comment (timgraham) [inline docs/ref/validators.txt]: It should be a separate block below this paragraph. See https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#documenting-new-features.

**Comment by timgraham** [comment]
merged in d58aaa24e31f10e56a7f05a4451cd06a3cc6e65d, thanks!

**Pull request closed without merging**